catchup/catchpointService.go

```diff
@@ -94,19 +94,14 @@ func MakeResumedCatchpointCatchupService(ctx context.Context, node CatchpointCat
 		stats: CatchpointCatchupStats{
 			StartTime: time.Now(),
 		},
-		node:           node,
-		ledgerAccessor: ledger.MakeCatchpointCatchupAccessor(l, log),
-		log:            log,
-		newService:     false,
-		net:            net,
-		ledger:         l,
-		config:         cfg,
-		blocksDownloadPeerSelector: makePeerSelector(
-			net,
-			[]peerClass{
-				{initialRank: peerRankInitialFirstPriority, peerClass: network.PeersPhonebookArchivers},
-				{initialRank: peerRankInitialSecondPriority, peerClass: network.PeersPhonebookRelays},
-			}),
+		node:                       node,
+		ledgerAccessor:             ledger.MakeCatchpointCatchupAccessor(l, log),
+		log:                        log,
+		newService:                 false,
+		net:                        net,
+		ledger:                     l,
+		config:                     cfg,
+		blocksDownloadPeerSelector: createPeerSelectorForCatchpoint(&cfg, net),
```

Comment:
this is a bit messy, since the createPeerSelectorForCatchpoint isn't going to be called from elsewhere.
why don't you create a method called `initDownloadPeerSelector`, can call it right after allocating the object ?
in that case, you don't need to pass any arguments since the `CatchpointCatchupService` object already has them all.

Reply:
sounds good.